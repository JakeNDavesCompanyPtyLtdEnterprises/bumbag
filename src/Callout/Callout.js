// @flow
import React, { type Element, type Node } from 'react';
import _Callout, { CalloutHeader } from './styled';
import { getUniqueId } from '../uniqueId';
import type { Palette } from '../types';

import CalloutContent from '../Card/CardContent';
import CalloutTitle from '../Card/CardTitle';
import Icon from '../Icon';

type Props = {
  a11yDescriptionId?: string,
  a11yTitleId?: string,
  children: Node,
  className?: string,
  hasTint?: boolean,
  icon?: string,
  palette?: Palette,
  title?: string | Element<any>
};

const Callout = ({ a11yDescriptionId, a11yTitleId, children, hasTint, icon, palette, title, ...props }: Props) => {
  const intentIcon = {
    primary: 'info-sign',
    warning: 'warning-sign',
    success: 'tick-circle',
    danger: 'error'
  };

  return (
    <_Callout
      a11yDescriptionId={a11yDescriptionId}
      a11yTitleId={a11yTitleId}
      hasTint={hasTint}
      palette={palette}
      {...props}
    >
      {title && (
        <CalloutHeader>
          {palette &&
            typeof intentIcon[palette] !== 'undefined' && (
              <Icon icon={icon ? icon : intentIcon[palette]} color={palette} marginRight="12px" size="medium" />
            )}
          {typeof title === 'string' ? (
            <CalloutTitle id={a11yTitleId} color={palette}>
              {title}
            </CalloutTitle>
          ) : (
            title
          )}
        </CalloutHeader>
      )}
      <CalloutContent id={a11yDescriptionId}>{children}</CalloutContent>
    </_Callout>
  );
};

Callout.defaultProps = {
  a11yDescriptionId: getUniqueId('Card'),
  a11yTitleId: getUniqueId('Card'),
  children: undefined,
  className: undefined,
  hasTint: undefined,
  icon: undefined,
  palette: 'text',
  title: undefined
};

export default Callout;
