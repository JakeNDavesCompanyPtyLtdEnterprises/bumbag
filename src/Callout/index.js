export { default } from './Callout';
export { default as Callout } from './Callout';
