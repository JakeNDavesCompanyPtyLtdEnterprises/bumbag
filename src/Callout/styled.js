// @flow

import styled, { css } from '../styled';
import { palette, theme } from 'styled-tools';
import CardCard from '../Card/CardCard';
import CardHeader from '../Card/CardHeader';
import CardContent from '../Card/CardContent';
import CardTitle from '../Card/CardTitle';
import CardFooter from '../Card/CardFooter';
import Icon from '../Icon';

const getTintAttributes = props => {
  const { type, hasTint } = props;

  if (!hasTint || !type) {
    return;
  }

  return css`
    background-color: ${type && palette(`${type}Tint`)(props)};
    color: ${type && palette(`${type}TintInverted`)(props)};
    fill: ${type && palette(`${type}TintInverted`)(props)};
    ${theme('fannypack.Callout.tint')};
  `;
};

export default styled(CardCard)`
  border: none;
  ${getTintAttributes};

  & {
    ${theme('fannypack.Callout.base')};
  }
`;

export const CalloutHeader = styled(CardHeader)`
  display: flex;
  justify-content: flex-start;

  & {
    ${theme('fannypack.Callout.Header.base')};
  }
`;

export const CalloutContent = styled(CardContent)`
  & {
    ${theme('fannypack.Callout.Content.base')};
  }
`;

export const CalloutTitle = styled(CardTitle)`
  & {
    ${theme('fannypack.Callout.Title.base')};
  }
`;

export const CalloutFooter = styled(CardFooter)`
  & {
    ${theme('fannypack.Callout.Footer.base')};
  }
`;

export const CalloutIcon = styled(Icon)`
  margin-right: ${theme('fannypack.layout.spacing.xsmall')}rem;

  & {
    ${theme('fannypack.Callout.Icon.base')};
  }
`;
