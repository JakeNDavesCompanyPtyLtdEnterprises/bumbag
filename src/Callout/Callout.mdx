
---
name: Callout
route: /components/callout
menu: Components
---

import { Playground, PropsTable } from 'docz';
import Callout from '../Callout';
import Button, { ActionButtons } from '../Button';
import Set from '../Set';

# Callout

## Import

`import { Callout } from 'fannypack'`

## Basic Usage

The `<Callout>` component uses `<Card>` internally, and is used to highlight important content to users.

If the content you are displaying is **brief**, **important** & requires the user's **immediate attention**, consider using the [`<Alert>` component](/components/alert)

You can render different intents for the callout using the `type` prop. A Callout can take a type of `"info"`, `"success"`, `"warning"` or `"danger"`.

<Playground>
  <Callout type="info" title="This is a title" marginBottom="20px">
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse diam ipsum, cursus id placerat congue, ultrices
    eget lectus. Duis posuere, lacus sed tristique commodo, sapien turpis mollis nunc, vestibulum consectetur lectus
    augue sit amet justo.
  </Callout>
  <Callout type="success" title="This is a title" marginBottom="20px">
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse diam ipsum, cursus id placerat congue, ultrices
    eget lectus. Duis posuere, lacus sed tristique commodo, sapien turpis mollis nunc, vestibulum consectetur lectus
    augue sit amet justo.
  </Callout>
  <Callout type="danger" title="This is a title" marginBottom="20px">
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse diam ipsum, cursus id placerat congue, ultrices
    eget lectus. Duis posuere, lacus sed tristique commodo, sapien turpis mollis nunc, vestibulum consectetur lectus
    augue sit amet justo.
  </Callout>
  <Callout type="warning" title="This is a title" marginBottom="20px">
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse diam ipsum, cursus id placerat congue, ultrices
    eget lectus. Duis posuere, lacus sed tristique commodo, sapien turpis mollis nunc, vestibulum consectetur lectus
    augue sit amet justo.
  </Callout>
</Playground>

## Footer

You can render a footer for the callout using the `footer` prop.

<Playground>
  <Callout type="info" title="This is a title" footer={<ActionButtons justifyContent="flex-end" width="100%" submitProps={{ palette: 'info' }} />}>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse diam ipsum, cursus id placerat congue, ultrices
    eget lectus. Duis posuere, lacus sed tristique commodo, sapien turpis mollis nunc, vestibulum consectetur lectus
    augue sit amet justo.
  </Callout>
</Playground>

## Close buttons

To add a close button, add the `showCloseButton` & `onClickClose` props.

**Note:** You can also add props to the close button with `closeButtonProps`. It takes [`<Button>`'s props](/components/button)

<Playground>
  <Callout type="info" title="This is a title" onClickClose={() => alert('Closed!')} showCloseButton marginBottom="20px">
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse diam ipsum, cursus id placerat congue, ultrices
    eget lectus. Duis posuere, lacus sed tristique commodo, sapien turpis mollis nunc, vestibulum consectetur lectus
    augue sit amet justo.
  </Callout>
</Playground>

## Tints

You can add a tint using the `hasTint` prop.

<Playground>
  <Callout hasTint type="info" title="This is a title" marginBottom="20px">
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse diam ipsum, cursus id placerat congue, ultrices
    eget lectus. Duis posuere, lacus sed tristique commodo, sapien turpis mollis nunc, vestibulum consectetur lectus
    augue sit amet justo.
  </Callout>
  <Callout hasTint type="success" title="This is a title" marginBottom="20px">
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse diam ipsum, cursus id placerat congue, ultrices
    eget lectus. Duis posuere, lacus sed tristique commodo, sapien turpis mollis nunc, vestibulum consectetur lectus
    augue sit amet justo.
  </Callout>
  <Callout hasTint type="danger" title="This is a title" marginBottom="20px">
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse diam ipsum, cursus id placerat congue, ultrices
    eget lectus. Duis posuere, lacus sed tristique commodo, sapien turpis mollis nunc, vestibulum consectetur lectus
    augue sit amet justo.
  </Callout>
  <Callout hasTint type="warning" title="This is a title" marginBottom="20px">
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse diam ipsum, cursus id placerat congue, ultrices
    eget lectus. Duis posuere, lacus sed tristique commodo, sapien turpis mollis nunc, vestibulum consectetur lectus
    augue sit amet justo.
  </Callout>
</Playground>

## Icon

You can also specify your own [`<Icon>` component](/components/icon) using the `icon` prop.

<Playground>
  <Callout icon="flame" title="This is a title">
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse diam ipsum, cursus id placerat congue, ultrices
    eget lectus. Duis posuere, lacus sed tristique commodo, sapien turpis mollis nunc, vestibulum consectetur lectus
    augue sit amet justo.
  </Callout>
</Playground>

## Theming

### Schema

```jsx
{
  base: string | Object,
  tint: string | Object,
  Content: {
    base: string | Object
  },
  Header: {
    base: string | Object
  },
  Footer: {
    base: string | Object
  },
  Title: {
    base: string | Object
  },
  Tint: {
    base: string | Object
  },
  Icon: {
    base: string | Object
  },
  Close: {
    base: string | Object,
    hover: string | Object
  }
}
```
