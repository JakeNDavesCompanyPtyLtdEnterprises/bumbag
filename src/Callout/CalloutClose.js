// @flow
import React from 'react';
import { CalloutClose as _CalloutClose } from './styled';
import Icon from '../Icon';
import VisuallyHidden from '../VisuallyHidden';

type Props = {
  onClickClose?: Function
};

const CalloutClose = ({ onClickClose, ...props }: Props) => (
  <_CalloutClose kind="link" onClick={onClickClose} {...props}>
    <VisuallyHidden>Close</VisuallyHidden>
    <Icon aria-hidden="true" icon="cross" />
  </_CalloutClose>
);

CalloutClose.defaultProps = {
  onClickClose: undefined
};

export default CalloutClose;
