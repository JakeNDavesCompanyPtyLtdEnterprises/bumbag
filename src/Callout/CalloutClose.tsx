import * as React from 'react';
// @ts-ignore
import PropTypes from 'prop-types';
import { ButtonProps as ReakitButtonProps } from 'reakit/ts/Button/Button';

import Icon from '../Icon';
import VisuallyHidden from '../VisuallyHidden';
import { Omit } from '../types';
import { CalloutClose as _CalloutClose } from './styled';

export interface LocalCalloutCloseProps {
  onClickClose?: () => void;
}
export type CalloutCloseProps = Omit<ReakitButtonProps, 'children'> & LocalCalloutCloseProps;

export const CalloutClose: React.FunctionComponent<LocalCalloutCloseProps> = ({ onClickClose, ...props }) => (
  <_CalloutClose kind="link" onClick={onClickClose} {...props}>
    <VisuallyHidden>Close</VisuallyHidden>
    <Icon a11yHidden icon="times" />
  </_CalloutClose>
);

CalloutClose.propTypes = {
  onClickClose: PropTypes.func
};
CalloutClose.defaultProps = {
  onClickClose: undefined
};

export default CalloutClose;
