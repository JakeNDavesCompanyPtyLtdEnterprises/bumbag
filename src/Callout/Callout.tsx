import * as React from 'react';

import { CardCardProps } from '../Card/CardCard';
import { Box, Flex } from '../primitives';
// @ts-ignore
import { getUniqueId } from '../uniqueId';
import _Callout, { CalloutContent, CalloutFooter, CalloutHeader, CalloutIcon, CalloutTitle } from './styled';
import CalloutClose, { CalloutCloseProps } from './CalloutClose';

export interface LocalCalloutProps {
  a11yDescriptionId?: string;
  a11yTitleId?: string;
  children: React.ReactNode;
  className?: string;
  closeButtonProps?: CalloutCloseProps;
  footer?: string | React.ReactElement<any>;
  hasTint?: boolean;
  icon?: string;
  onClickClose?: CalloutCloseProps['onClickClose'];
  showCloseButton?: boolean;
  title?: string | React.ReactElement<any>;
  type?: string;
}
export type CalloutProps = CardCardProps & LocalCalloutProps;

export const Callout: React.FunctionComponent<LocalCalloutProps> = ({
  a11yDescriptionId,
  a11yTitleId,
  children,
  closeButtonProps,
  footer,
  hasTint,
  icon,
  onClickClose,
  showCloseButton,
  title,
  type,
  ...props
}) => {
  return (
    <_Callout a11yDescriptionId={a11yDescriptionId} a11yTitleId={a11yTitleId} hasTint={hasTint} type={type} {...props}>
      <Flex>
        {(type || icon) && (
          <CalloutTitle>
            {/*
              // @ts-ignore */}
            <CalloutIcon icon={(type || icon) as string} color={type} />
          </CalloutTitle>
        )}
        <Box>
          {title && (
            <CalloutHeader>
              {typeof title === 'string' ? <CalloutTitle id={a11yTitleId}>{title}</CalloutTitle> : title}
            </CalloutHeader>
          )}
          <CalloutContent id={a11yDescriptionId}>{children}</CalloutContent>
          {footer && <CalloutFooter>{footer}</CalloutFooter>}
        </Box>
      </Flex>
      {showCloseButton && <CalloutClose onClickClose={onClickClose} {...closeButtonProps} />}
    </_Callout>
  );
};

Callout.defaultProps = {
  a11yDescriptionId: getUniqueId('Callout'),
  a11yTitleId: getUniqueId('Callout'),
  children: undefined,
  className: undefined,
  closeButtonProps: {},
  footer: undefined,
  hasTint: false,
  icon: undefined,
  onClickClose: undefined,
  showCloseButton: false,
  title: undefined,
  type: undefined
};

export default Callout;
