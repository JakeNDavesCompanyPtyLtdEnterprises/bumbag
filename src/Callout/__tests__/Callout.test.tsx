import * as React from 'react';
import render from '../../_utils/tests/render';
import Callout from '../Callout';

it('renders correctly', () => {
  const { container } = render(<Callout>Callout</Callout>);
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly with a custom element', () => {
  const { container } = render(
    <Callout>
      <label>Callout</label>
    </Callout>
  );
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly when a type prop is set', () => {
  const { container } = render(<Callout type="info">Callout</Callout>);
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly when a title prop is set', () => {
  const { container } = render(<Callout title="Title">Callout</Callout>);
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly when a footer prop is set', () => {
  const { container } = render(<Callout footer={<div>footer</div>}>Callout</Callout>);
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly when a hasTint prop is set', () => {
  const { container } = render(<Callout hasTint>Callout</Callout>);
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly when a close button is present', () => {
  const { container } = render(
    <Callout title="Title" onClickClose={() => {}} showCloseButton>
      Callout
    </Callout>
  );
  expect(container.firstChild).toMatchSnapshot();
});
