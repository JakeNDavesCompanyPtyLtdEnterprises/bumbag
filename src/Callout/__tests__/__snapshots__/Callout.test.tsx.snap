// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c3 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c3:focus:not(:focus-visible) {
  outline: none;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  background-color: white;
  border-radius: 3px;
  box-shadow: 0px 2px 4px 0px rgba(0,0,0,0.1),0px 0px 0px 1px rgba(0,0,0,0.1);
}

.c1 {
  padding: 1.5rem;
}

.c0 {
  position: relative;
}

<div
  aria-describedby="Callout-1"
  aria-labelledby="Callout-2"
  class="c0 c1 c2 c3"
>
  <div
    class="c4 c3"
  >
    <div
      class="c3"
    >
      <div
        class="c3"
        id="Callout-1"
      >
        Callout
      </div>
    </div>
  </div>
</div>
`;

exports[`renders correctly when a close button is present 1`] = `
.c3 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c3:focus:not(:focus-visible) {
  outline: none;
}

.c11 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: #fcfcfc;
  color: inherit;
}

.c11:focus:not(:focus-visible) {
  outline: none;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c14 {
  fill: inherit;
  position: relative;
  font-size: 1em;
  height: 1em;
  width: 1em;
}

.c10 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fcfcfc;
  border-radius: 4px;
  color: #435a6f;
  fill: #435a6f;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  min-height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: 1px solid #b3b3b3;
}

.c10:focus {
  outline: unset;
  z-index: 2;
  box-shadow: #444bc9 0px 0px 0px 2px;
}

.c10[disabled] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

.c10 {
  border: 0;
  background: unset;
  color: #435a6f;
  fill: #435a6f;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c10:hover {
  color: #000;
  fill: #000;
}

.c8 {
  font-weight: 700;
  line-height: 1.2;
}

.c8:not(:last-child) {
  margin-bottom: 1rem;
}

.heading.c8 + .sub-heading {
  margin-top: -1rem;
}

h1.c8 {
  font-size: 3rem;
}

h2.c8 {
  font-size: 2.5rem;
}

h3.c8 {
  font-size: 2rem;
}

h4.c8 {
  font-size: 1.5rem;
}

h5.c8 {
  font-size: 1.25rem;
}

h6.c8 {
  font-size: 1rem;
}

.c8 .c13 {
  top: 0.15em;
}

.c8 {
  font-weight: 600;
}

.c2 {
  background-color: white;
  border-radius: 3px;
  box-shadow: 0px 2px 4px 0px rgba(0,0,0,0.1),0px 0px 0px 1px rgba(0,0,0,0.1);
}

.c12 {
  border: 0px;
  -webkit-clip: rect(0px,0px,0px,0px);
  clip: rect(0px,0px,0px,0px);
  height: 1px;
  width: 1px;
  margin: -1px;
  padding: 0px;
  overflow: hidden;
  position: absolute;
}

.c1 {
  padding: 1.5rem;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-bottom: 1em;
  position: relative;
  width: 100%;
}

.c7 {
  margin-bottom: 0px;
}

.c0 {
  position: relative;
}

.c9 {
  position: absolute;
  right: 0.8rem;
  top: 0.8rem;
  width: 30px;
  height: 30px;
  padding: 0;
  text-align: right;
}

.c9:hover {
  background-color: #f2f2f2;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

<div
  aria-describedby="Callout-1"
  aria-labelledby="Callout-2"
  class="c0 c1 c2 c3"
>
  <div
    class="c4 c3"
  >
    <div
      class="c3"
    >
      <div
        class="c5 c6 c3"
      >
        <h5
          class="c7 sub-heading c8 c3"
          id="Callout-2"
        >
          Title
        </h5>
      </div>
      <div
        class="c3"
        id="Callout-1"
      >
        Callout
      </div>
    </div>
  </div>
  <button
    class="c9 c10 c11"
    kind="link"
    type="button"
  >
    <div
      class="c12 c3"
    >
      Close
    </div>
    <svg
      aria-hidden="true"
      class="c13 c14 c3"
      role="img"
      size="1"
      style="size: 1px;"
      viewBox="0 0 352 512"
    >
      <path
        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</div>
`;

exports[`renders correctly when a footer prop is set 1`] = `
.c3 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c3:focus:not(:focus-visible) {
  outline: none;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  background-color: white;
  border-radius: 3px;
  box-shadow: 0px 2px 4px 0px rgba(0,0,0,0.1),0px 0px 0px 1px rgba(0,0,0,0.1);
}

.c1 {
  padding: 1.5rem;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 1rem;
  width: 100%;
}

.c0 {
  position: relative;
}

<div
  aria-describedby="Callout-1"
  aria-labelledby="Callout-2"
  class="c0 c1 c2 c3"
>
  <div
    class="c4 c3"
  >
    <div
      class="c3"
    >
      <div
        class="c3"
        id="Callout-1"
      >
        Callout
      </div>
      <div
        class="c5 c3"
      >
        <div>
          footer
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders correctly when a hasTint prop is set 1`] = `
.c3 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c3:focus:not(:focus-visible) {
  outline: none;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  background-color: white;
  border-radius: 3px;
  box-shadow: 0px 2px 4px 0px rgba(0,0,0,0.1),0px 0px 0px 1px rgba(0,0,0,0.1);
}

.c1 {
  padding: 1.5rem;
}

.c0 {
  position: relative;
}

<div
  aria-describedby="Callout-1"
  aria-labelledby="Callout-2"
  class="c0 c1 c2 c3"
>
  <div
    class="c4 c3"
  >
    <div
      class="c3"
    >
      <div
        class="c3"
        id="Callout-1"
      >
        Callout
      </div>
    </div>
  </div>
</div>
`;

exports[`renders correctly when a title prop is set 1`] = `
.c3 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c3:focus:not(:focus-visible) {
  outline: none;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c8 {
  font-weight: 700;
  line-height: 1.2;
}

.c8:not(:last-child) {
  margin-bottom: 1rem;
}

.heading.c8 + .sub-heading {
  margin-top: -1rem;
}

h1.c8 {
  font-size: 3rem;
}

h2.c8 {
  font-size: 2.5rem;
}

h3.c8 {
  font-size: 2rem;
}

h4.c8 {
  font-size: 1.5rem;
}

h5.c8 {
  font-size: 1.25rem;
}

h6.c8 {
  font-size: 1rem;
}

.c8 .c9 {
  top: 0.15em;
}

.c8 {
  font-weight: 600;
}

.c2 {
  background-color: white;
  border-radius: 3px;
  box-shadow: 0px 2px 4px 0px rgba(0,0,0,0.1),0px 0px 0px 1px rgba(0,0,0,0.1);
}

.c1 {
  padding: 1.5rem;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-bottom: 1em;
  position: relative;
  width: 100%;
}

.c7 {
  margin-bottom: 0px;
}

.c0 {
  position: relative;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

<div
  aria-describedby="Callout-1"
  aria-labelledby="Callout-2"
  class="c0 c1 c2 c3"
>
  <div
    class="c4 c3"
  >
    <div
      class="c3"
    >
      <div
        class="c5 c6 c3"
      >
        <h5
          class="c7 sub-heading c8 c3"
          id="Callout-2"
        >
          Title
        </h5>
      </div>
      <div
        class="c3"
        id="Callout-1"
      >
        Callout
      </div>
    </div>
  </div>
</div>
`;

exports[`renders correctly when a type prop is set 1`] = `
.c3 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c3:focus:not(:focus-visible) {
  outline: none;
}

.c10 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
  color: #35709e !important;
}

.c10:focus:not(:focus-visible) {
  outline: none;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c9 {
  fill: #35709e;
  position: relative;
  font-size: 1em;
  height: 1em;
  width: 1em;
}

.c6 {
  font-weight: 700;
  line-height: 1.2;
}

.c6:not(:last-child) {
  margin-bottom: 1rem;
}

.heading.c6 + .sub-heading {
  margin-top: -1rem;
}

h1.c6 {
  font-size: 3rem;
}

h2.c6 {
  font-size: 2.5rem;
}

h3.c6 {
  font-size: 2rem;
}

h4.c6 {
  font-size: 1.5rem;
}

h5.c6 {
  font-size: 1.25rem;
}

h6.c6 {
  font-size: 1rem;
}

.c6 .c8 {
  top: 0.15em;
}

.c6 {
  font-weight: 600;
}

.c2 {
  background-color: white;
  border-radius: 3px;
  box-shadow: 0px 2px 4px 0px rgba(0,0,0,0.1),0px 0px 0px 1px rgba(0,0,0,0.1);
}

.c1 {
  padding: 1.5rem;
}

.c5 {
  margin-bottom: 0px;
}

.c0 {
  position: relative;
}

.c7 {
  margin-right: 1rem;
}

<div
  aria-describedby="Callout-1"
  aria-labelledby="Callout-2"
  class="c0 c1 c2 c3"
  type="info"
>
  <div
    class="c4 c3"
  >
    <h5
      class="c5 sub-heading c6 c3"
    >
      <svg
        class="c7 c8 c9 c10"
        color="info"
        role="img"
        size="1"
        style="size: 1px;"
        viewBox="0 0 512 512"
      >
        <path
          d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </h5>
    <div
      class="c3"
    >
      <div
        class="c3"
        id="Callout-1"
      >
        Callout
      </div>
    </div>
  </div>
</div>
`;

exports[`renders correctly with a custom element 1`] = `
.c3 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c3:focus:not(:focus-visible) {
  outline: none;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  background-color: white;
  border-radius: 3px;
  box-shadow: 0px 2px 4px 0px rgba(0,0,0,0.1),0px 0px 0px 1px rgba(0,0,0,0.1);
}

.c1 {
  padding: 1.5rem;
}

.c0 {
  position: relative;
}

<div
  aria-describedby="Callout-1"
  aria-labelledby="Callout-2"
  class="c0 c1 c2 c3"
>
  <div
    class="c4 c3"
  >
    <div
      class="c3"
    >
      <div
        class="c3"
        id="Callout-1"
      >
        <label>
          Callout
        </label>
      </div>
    </div>
  </div>
</div>
`;
